components/modals/addQuestionModal/AddQuestionModal.tsx

```diff
@@ -1,248 +1,164 @@
 
+import { AddQuestionModalContent } from './AddQuestionModalContent';
+import { AddQuestionModalFooter } from './AddQuestionModalFooter';
 import styles from './addQuestionModal.module.scss';
 
-interface AddQuestionModalState {
-  technology?: TechnologyKey;
-  level?: LevelKey;
-  questionText: string;
-  isLoading: boolean;
-  valid: boolean;
-  originalQuestion?: Question;
-}
+const reportEvent = (action: string) => globalReportEvent(action, 'Nowe pytanie warstwa');
 
 interface AddQuestionModalOwnProps {
   originalQuestion?: Question;
 }
 
-type AddQuestionModalProps = AddQuestionModalOwnProps &
-  CommonModalProps &
-  ReturnType<typeof mapStateToProps> &
-  typeof mapDispatchToProps;
-
-class AddQuestionModalComponent extends React.PureComponent<
-  AddQuestionModalProps,
-  AddQuestionModalState
-> {
-  static getDerivedStateFromProps: React.GetDerivedStateFromProps<
-    AddQuestionModalProps,
-    AddQuestionModalState
-  > = (props, state) => {
-    if (!props.originalQuestion || isEqual(props.originalQuestion, state.originalQuestion)) {
-      return null;
-    }
-
-    const { originalQuestion } = props;
-
-    return {
-      originalQuestion,
-      level: originalQuestion._levelId,
-      technology: originalQuestion._categoryId,
-      questionText: originalQuestion.question,
-    };
-  };
+type AddQuestionModalProps = AddQuestionModalOwnProps & CommonModalProps;
+
+export const AddQuestionModal = memo(
+  forwardRef<HTMLDivElement, AddQuestionModalProps>(({ onClose, originalQuestion }, ref) => {
+    const [editedQuestion, setEditedQuestion] = useState<Question>();
+    const [questionText, setQuestionText] = useState('');
+    const [level, setLevel] = useState<LevelKey>();
+    const [technology, setTechnology] = useState<TechnologyKey>();
+    const [isLoading, setIsLoading] = useState(false);
+    const [valid, setValid] = useState(false);
+
+    const dispatch = useDispatch();
+
+    const isValid = useCallback(() => Boolean(level && technology && questionText.trim()), [
+      level,
+      questionText,
+      technology,
+    ]);
+
+    const handleClose: CommonModalProps['onClose'] = useCallback(
+      (args) => {
+        if (args.reason === 'cancel') {
+          reportEvent('Anuluj');
+        } else if (args.reason === 'submit') {
+          reportEvent('Dodaj pytanie');
+        } else {
+          reportEvent('Zamknij');
+        }
+        return onClose(args);
+      },
+      [onClose]
+    );
 
-  state: AddQuestionModalState = { questionText: '', isLoading: false, valid: false };
+    const onCancelClick: React.MouseEventHandler<HTMLButtonElement> = useCallback(
+      (e) => {
+        handleClose({ reason: 'cancel', event: e });
+      },
+      [handleClose]
+    );
 
-  componentDidMount() {
-    this.reportEvent('Wyświetlenie');
-    this.setState((state) => ({ valid: this.isValid(state) }));
-  }
+    const handleChangeTechnology: React.ChangeEventHandler<HTMLSelectElement> = useCallback((e) => {
```

Comment:
Maybe consider to create alias and reused it? Because you're using it couple of times. 
In line 63 and 68.
```
type something = React.ChangeEventHandler<HTMLSelectElement>;
```